src/pages/search/SearchPage.tsx

```
@@ -180,16 +185,62 @@ export default function SearchPage() {
     useEffect(() => {
         getFoodTypes()
             .then(setFoodTypes)
             .catch(() => setFoodTypesError("음식 종류 목록을 불러오지 못했습니다."));
     }, []);
 
+    // 로그인 사용자의 인증 동네를 가져와서 RegionSelect 표시값 초기화
+    // (실제 검색 region 파라미터는 빈 문자열로 유지 → BE 가 user.region 자동 적용)
+    useEffect(() => {
+        // 로그아웃(accessToken null) 시 라벨/깃발 초기화
+        if (!accessToken) {
+            setSelectedRegionName("전체");
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/src/pages/search/SearchPage.tsx:209:13
  207 |         // 로그아웃(accessToken null) 시 라벨/깃발 초기화
  208 |         if (!accessToken) {
> 209 |             setSelectedRegionName("전체");
      |             ^^^^^^^^^^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  210 |             setHasManuallySelectedRegion(false);
  211 |             return;
  212 |         }

(react-hooks/set-state-in-effect)